cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/index.js

```diff
@@ -55,26 +55,34 @@ export default class ColumnsTab extends Component {
     this.clearAllColumns = this.clearAllColumns.bind(this);
     this.selectAllColumns = this.selectAllColumns.bind(this);
     this.setSelect = this.setSelect.bind(this);
+  }
 
+  componentDidMount() {
+    this._isMounted = true;
```

Comment:
Do we need this flag? I remember seeing this in some other component as well. Shouldn't the CDM be called only once when the component is mounted? If this is getting called multiple times then we are un-necessarily unmounting and remounting react components which would be the source of the problem.
This looks more like a hack for us to avoid it. Is there any specific reason why we are adding this flag?

Reply:
Hhmm this seems a little hacky. The subscriptions are snapshotted which causes this issue: https://redux.js.org/api/store#subscribe-listener. The right fix would be to use react-redux and use connected components. Manually subscribing and unsubscribing will cause this issue though. Ok may not be in this PR but we need to change across UI to use `connect`d components so that we don't have to worry about it.